packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -294,10 +304,15 @@ function processModelProperties(
     if (!hasDiscriminator) {
       newValue.properties.push({ ...discriminatorInfo });
     }
-    newValue.alias = `${newValue.name}Parent`;
-    newValue.isPolyBaseModel = true;
-    discriminatorInfo?.aliases.push(`${newValue.alias}`);
-    newValue.aliasType = discriminatorInfo?.aliases.join(" | ");
+    if (newValue.name && newValue.name !== "") {
```

Comment:
`if(newValue.name)`  or `if(Boolean(newValue.name))`  should be enough to check for undefined or empty string

Reply:
Can we also add some docs describing what this check means?